### PR TITLE
[#21] Allow to filter keywords by name

### LINKED ISSUE #21: [Backend] As a user, I can filter the list of keywords by name
## Why

The user wants to filter their uploaded keywords.

## Acceptance Criteria

The server allows filtering queryset by these params by keyword name, with using `%like%` query

### PR DESCRIPTION
**Pull request by Nihisil**
Resolves #21

## What happened 👀

Added feature to do wildcard search by keyword names.

## Proof Of Work 📹

Tests are green!

### CODE REVIEW

File: test/google_scraping/dashboard/keywords_test.exs [hunk 1]
```diff
@@ -4,14 +4,20 @@ defmodule GoogleScraping.Dashboard.KeywordsTest do
...
 
-      assert Keywords.list_keywords(user.id) == [keyword]
+      assert Keywords.list_keywords(keyword.user_id, nil) == [keyword]
+    end
+
+    test "with provided search phrase, returns filtered queryset" do
+      user = insert(:user)
+      _cat_keyword = insert(:keyword, name: "cat", user_id: user.id)
+      dog_keyword = insert(:keyword, name: "dog", user_id: user.id)
+      assert Keywords.list_keywords(user.id, %{"query" => "og"}) == [dog_keyword]
```
Comment (junan) [inline test/google_scraping/dashboard/keywords_test.exs:20]:
```suggestion

      assert Keywords.list_keywords(user.id, %{"query" => "og"}) == [dog_keyword]
```

**Review by junan — APPROVED** [review]

File: lib/google_scraping/dashboard/queries/keyword_query.ex [hunk 2]
```diff
@@ -6,5 +6,8 @@ defmodule GoogleScraping.Dashboard.Queries.KeywordQuery do
   @doc """
   Returns the keywords list for the given user_id
   """
-  def user_keywords_query(user_id), do: where(Keyword, [k], k.user_id == ^user_id)
+  def user_keywords(user_id, search_phrase \\ nil) do
+    wildcard_search = "%#{search_phrase}%"
+    Keyword |> where([k], k.user_id == ^user_id) |> where([k], ilike(k.name, ^wildcard_search))
```
Comment (bterone) [inline lib/google_scraping/dashboard/queries/keyword_query.ex:11]: Minor comment but piping vertically makes it easier to read

```suggestion
    wildcard_search = "%#{search_phrase}%"

    Keyword 
    |> where([k], k.user_id == ^user_id) 
    |> where([k], ilike(k.name, ^wildcard_search))
```

File: lib/google_scraping/dashboard/keywords.ex [hunk 3]
```diff
@@ -16,11 +16,16 @@ defmodule GoogleScraping.Dashboard.Keywords do
...
+      iex> list_keywords(user_id, nil)
       [%Keyword{}, ...]
 
+      iex> list_keywords(user_id, %{"query" => "dog"})
+      [%Keyword{}, ...]
   """
-  def list_keywords(user_id), do: Repo.all(KeywordQuery.user_keywords_query(user_id))
+  def list_keywords(user_id, params \\ %{}) do
+    search_phrase = get_in(params, ["query"])
+    Repo.all(KeywordQuery.user_keywords(user_id, search_phrase))
```
Comment (bterone) [inline lib/google_scraping/dashboard/keywords.ex:27]: Another place where we can pipe vertically 😄 

```suggestion
    user_id
    |> KeywordQuery.user_keywords(search_phrase)
    |> Repo.all()
```

**Review by bterone — APPROVED** [review]
Just some stylistic comments, otherwise LGTM! 👍

**Pull request merged**